framework/src/st_things/things_stack/framework/things_data_manager.c

```diff
@@ -2202,26 +2104,24 @@ int dm_register_resource(things_server_builder_s *p_builder)
 
 		THINGS_LOG_D(TAG, "SINGLE RESOURCE(S) CNT : %d", device->sig_cnt);
 		for (int capa_num = 0; capa_num < device->sig_cnt; capa_num++) {
-			resource = &device->single[capa_num];
+			struct things_resource_info_s *resource = &device->single[capa_num];
 			device->pchild_resources[n_count_of_children++] = register_resource(p_builder, resource, id);
 		}					// End of for single resource registration loop
 
 		// Create the Device(/oic/d) and Platform (/oic/p)
 		// ex) Aircon Multi-Model
-		if (device_num < 1) {
-			// for hosting device
-			THINGS_LOG_D(TAG, "REGISTERGING DEVICE INFO. TO DEVICE(/oic/d).");
-			p_builder->set_device_info(p_builder, device->name, device->type);
-			THINGS_LOG_D(TAG, "REGISTERGING DEVICE INFO. TO PLATFORM(/oic/p).");
-			p_builder->set_platform_info(p_builder, device->model_num,	// gDeviceModel,
-											device->ver_p,	// gPlatformVersion,
-											device->ver_os,	// gOSVersion,
-											device->ver_hw,	// gHWVersions,
-											device->ver_fw,	// gFWVersions,
-											device->vid,	// gVenderId
-											device->mnid,	// manufacturer_name
-											device->manufacturer_url);	// manufacturer_url
-		}
+		// for hosting device
+		THINGS_LOG_D(TAG, "REGISTERGING DEVICE INFO. TO DEVICE(/oic/d).");
+		p_builder->set_device_info(p_builder, device->name, device->type);
+		THINGS_LOG_D(TAG, "REGISTERGING DEVICE INFO. TO PLATFORM(/oic/p).");
```

Comment:
- any particular reason for removing if condition ?

- please check if formatting can be improved for readability ?

Reply:
- device_num is always <1 and hence there is no need to check.

- Sure. Formatting can be handled separately.